packages/foam-vscode/src/features/open-random-note.ts

```diff
@@ -0,0 +1,29 @@
+import { Foam } from 'foam-core';
+import { ExtensionContext, commands, window } from 'vscode';
+import { FoamFeature } from '../types';
+import { focusNote } from '../utils';
+
+const feature: FoamFeature = {
+  activate: (context: ExtensionContext, foamPromise: Promise<Foam>) => {
+    context.subscriptions.push(
+      commands.registerCommand('foam-vscode.open-random-note', async () => {
+        const foam = await foamPromise;
+        const currentFile = window.activeTextEditor?.document.uri.path;
+        const notes = foam.notes
+          .getNotes()
+          .map(note => note.uri.path)
+          .filter(notePath => notePath !== currentFile);
```

Comment:
instead of filtering the whole array of notes to remove one, which can be expensive (O(n)) as your knowledge base grows, we could optimistically pick a note, and if it's the current one pick another random note, wdyt?
You can still check that there is more than 1 note in the array below to keep that check.

Reply:
That's a great idea